t/frame.c

```diff
@@ -89,6 +89,26 @@ static void test_ack_decode(void)
         ok(decoded.smallest_acknowledged == 0x34 - 1 - 2 - 3 - 4 - 5 + 1);
     }
 
+    { /* Bogus ACK Frame larger than the internal buffer */
+        uint8_t pat[1024], *pos = pat;
+        const uint8_t *src = pat;
+        int i;
+        quicly_ack_frame_t decoded;
+        pos = quicly_encodev(pos, 0x84D0);
+        pos = quicly_encodev(pos, 0);
+        pos = quicly_encodev(pos, QUICLY_MAX_ACK_RANGE_COUNT + 1);
+        pos = quicly_encodev(pos, 8);
+        for (i = 0; i <= QUICLY_MAX_ACK_RANGE_COUNT; ++i) {
+            pos = quicly_encodev(pos, i); // gap
+            pos = quicly_encodev(pos, 1); // ack-range
+        }
+        ok(quicly_decode_ack_frame(&src, pos, &decoded, 0) == 0);
+        ok(decoded.largest_acknowledged == 0x84D0);
+        ok(decoded.ack_delay == 0);
+        ok(decoded.num_gaps == QUICLY_MAX_ACK_RANGE_COUNT + 1);
```

Comment:
I am not sure if this is the correct behavior.

I think what we the decode function should do is to pretend as if it saw a ACK frame containing MAX_ACK_RANGE_COUNT ranges, regardless of how many ranges the frame contained.

Otherwise, the user of `quicly_ack_frame_t` cannot use `num_gaps` to determine how many ranges and gaps is contained in that structure. Please see `handle_ack_frame` for how the value is used.

The same argument goes to `smallest_acknowledged` too. I'd appreciate it if you could check that all other members act the way described above.

Reply:
Thank you for catching this. You're absolutely right that this unit test is testing the wrong behavior, and is incomplete for not testing other members (especially smallest_acknowledged). 6114e850fef037631587fae23e27f3c68789e0ab is my attempt at addressing these feedback.